middleware/spoolman/client.py

```
@@ -43,6 +43,20 @@ def _fetch_all_spools(self) -> None:
 
+    def refresh(self) -> None:
+        """Public wrapper around _fetch_all_spools for explicit cache priming."""
+        self._fetch_all_spools()
+
+    def get_spool_by_id(self, spool_id: int) -> Optional[dict]:
+        """Fetch a single spool directly from Spoolman by ID. Returns None on failure."""
+        try:
+            response = requests.get(f"{self.base_url}/api/v1/spool/{spool_id}", timeout=5)
+            response.raise_for_status()
+            return response.json()
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)